### PR TITLE
36 update and cleanup docstrings

### PR DESCRIPTION
**Pull request by asmitakarandikar**
# Pull request

**Resolves:** #36

### CODE REVIEW

**Review by Viraj-Ganguli — APPROVED** [review]
Pranav and I took a look and it looks good. We appreciate the verbosity!

File: darkmoon/api/v1/metadata/schema.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
-"""Imports the modules/classes Field, BaseModel, and Optional."""
-
 from beanie import PydanticObjectId
 from pydantic import BaseModel, Field
 
+"""Import module/class PydanticObjectID, BaseModel, and Field"""
```
Comment (DoodlePooge) [inline darkmoon/api/v1/metadata/schema.py:4]: This did not need to be moved. In fact, the docstring itself at the top should more so explain the purpose of the file itself. Imports don't need docstrings

File: darkmoon/api/v1/metadata/views.py [hunk 2]
```diff
@@ -14,19 +9,14 @@
...
-####################
-# GLOBAL VARIABLES #
-####################
-
-router = APIRouter(prefix="/metadata", tags=["metadata"])
-
-###########
-# CLASSES #
+"""Import various modules and dependencies such as bson, PydanticObjectId from
+ beanie, APIRouter, Depends, HTTPException, Query from fastapi, AsyncIOMotorCollection
```
Comment (DoodlePooge) [inline darkmoon/api/v1/metadata/views.py:13]: Again not necessary

File: darkmoon/api/v1/metadata/views.py [hunk 3]
```diff
@@ -1,9 +1,4 @@
-"""This is the router file."""
-
-###########
-# IMPORTS #
-###########
-
+"""This is the views.py file."""
```
Comment (DoodlePooge) [inline darkmoon/api/v1/metadata/views.py:1]: If we are keeping these docstrings at the top, they need to be more useful than stating the name of the file.

File: darkmoon/api/v1/metadata/views.py [hunk 4]
```diff
@@ -38,7 +28,7 @@ async def list_metadata(
     page: int = Query(0, ge=0, description="The page to iterate to."),
     length: int = Query(10, ge=1, le=500),
 ) -> list[MetadataEntity]:
-    """Return list of metadata that matches the parameters in the database.
+    """List of metadata that matches the parameters in the database.
```
Comment (DoodlePooge) [inline darkmoon/api/v1/metadata/views.py:31]: This endpoint is not a list, but it grabs a list. You can say something like Get list

File: darkmoon/api/v1/metadata/views.py [hunk 5]
```diff
@@ -122,7 +112,7 @@ async def upload_metadata(
     file: Metadata,
     collection: AsyncIOMotorCollection = Depends(get_file_metadata_collection),
 ) -> None:
-    """Fast API POST function for incoming files.
+    """Fast API post function for incoming files.
```
Comment (DoodlePooge) [inline darkmoon/api/v1/metadata/views.py]: "Adds metadata from files to the database"
Saying that this is a FastAPI POST function is redundant.

File: darkmoon/api/v1/metadata/views.py [hunk 6]
```diff
@@ -122,7 +112,7 @@ async def upload_metadata(
...
```
Comment (DoodlePooge) [inline darkmoon/api/v1/metadata/views.py]: Docstrings should also provide any exceptions the function raises.

File: darkmoon/app.py [hunk 7]
```diff
@@ -4,19 +4,29 @@
...
 from darkmoon.settings import Settings
 
+app = FastAPI()
+
+templates = Jinja2Templates(directory="templates")
+
+
+@app.get("/items/{id}", response_class=HTMLResponse)
+async def read_item(request: Request, id: str) -> Response:
+    """Return the FastAPI connection."""
```
Comment (DoodlePooge) [inline darkmoon/app.py:22]: Is this accurate to the function's purpose?

File: darkmoon/cli/main.py
Comment (DoodlePooge) [inline darkmoon/cli/main.py:1]: Again, the top of the file docstring should give a developer an idea of what this file provides to the project. What is the main.py file? why is it?

File: darkmoon/cli/main.py [hunk 8]
```diff
@@ -39,7 +39,7 @@ def get_metadata(
         ),
     ],
 ) -> None:
-    """Call all of the metadata functions and send data to api endpoint."""
+    """Call the get metadata function from utils.py and send data to api endpoint."""
```
Comment (DoodlePooge) [inline darkmoon/cli/main.py:42]: For all of these, you don't need to say exactly what is run. The docstring should give a general purpose, parameters, raises (exceptions), and returns

File: requirements.txt
Comment (DoodlePooge) [inline requirements.txt:1]: please delete this

File: pyproject.toml [hunk 9]
```diff
@@ -22,6 +22,7 @@ uvicorn = { extras = ["standard"], version = "^0.22.0" }
 python-magic = "^0.4.27"
 beanie = "^1.19.2"
 typer = { extras = ["all"], version = "^0.9.0" }
+jinja2 = "^3.1.2"
```
Comment (DoodlePooge) [inline pyproject.toml:25]: Why is this added in the docstring ticket?

Comment (DoodlePooge) [reply pyproject.toml:25]: it's not bad but it means you're pulling changes from other places which is just a bad habit

**Review by DoodlePooge — CHANGES_REQUESTED** [review]

**Comment by DoodlePooge** [comment]
There are also still unverified commits, these need to be fixed or we can't merge it in.

**Pull request closed without merging**